### PR TITLE
update charm dep

### PR DESCRIPTION
**Pull request by nammn**
## Description of change

This references to this pr https://github.com/juju/charm/pull/297.
tl;dr change:
- refactoring of the charm version generation code
- only throw a warn on err events, else just log
- generates git version of subdir of git folders

### CODE REVIEW

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
these test fail https://github.com/juju/juju/blob/6ea20273f7c540d93b1075d77aafe39e70770b33/api/client_test.go#L428-L427 
because with the newest change git version is generated 
```
	expected, err := ioutil.ReadFile(ch.Path)
```
reads the file with the version file

The 
```
	reader, err := client.OpenCharm(curl)
```
loads this from the url which has the new git generated version.

**Comment by nammn** [comment]
@SimonRichardson ^ wdyt?

**Comment by SimonRichardson** [comment]
So I've got a forth option.

Ensure that the isolation suite does isolate against the git repo of juju/juju being there. You can do this in theory by changing what files are located in the home directory - https://github.com/juju/testing/blob/master/home.go

Note: the consuite composes the https://github.com/juju/testing/blob/master/home.go#L98

**Comment by nammn** [comment]
Thanks for the update!
Update from discussion out of daily.

5 (or adding to 4). Option: copying it to the tmp dir we create and use it as deploy dir.

**Comment by nammn** [comment]
remote hang up..

!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
30 min dep ..
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!! remote blocking

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
!!build!!

**Comment by nammn** [comment]
$$merge$$

**Comment by nammn** [comment]
$$merge$$ 

```17:27:54 	(1) failed to list versions for https://code.googlesource.com/gocloud: fatal: unable to access 'https://code.googlesource.com/gocloud/': gnutls_handshake() failed: The TLS```
We need to do something about this in the future..

File: testcharms/charm.go [hunk 1]
```diff
@@ -8,14 +8,19 @@ package testcharms
 import (
 	"archive/zip"
 	"bytes"
+	"fmt"
+	"github.com/juju/utils/fs"
```
Comment (manadart) [inline testcharms/charm.go]: Bad grouping here... for next time.

Comment (nammn) [reply testcharms/charm.go]: Sorry, this somehow got lost. Next time :+1:

**Comment by SimonRichardson** [comment]
If you feel daring, you could try https://github.com/SimonRichardson/stanzas

This organises imports how juju likes them.

**Pull request merged**